Libraries/Opc.Ua.Security.Certificates/Extensions/X509SubjectAltNameExtension.cs

```diff
@@ -0,0 +1,446 @@
+        /// </summary>
+        /// <value>The domain names.</value>
+        public IReadOnlyList<string> DomainNames
+        {
+            get
+            {
+                EnsureDecoded();
+                return m_domainNames.AsReadOnly();
+            }
+        }
+
+        /// <summary>
+        /// Gets the IP addresses.
+        /// </summary>
+        /// <value>The IP addresses.</value>
+        public IReadOnlyList<string> IPAddresses
+        {
+            get
+            {
+                EnsureDecoded();
+                return m_ipAddresses.AsReadOnly();
+            }
+        }
+        #endregion
+
+        #region Private Methods
+        /// <summary>
+        /// Create a normalized IPv4 or IPv6 address from a 4 byte or 16 byte array.
+        /// </summary>
+        private string IPAddressToString(byte[] encodedIPAddress)
+        {
+            try
+            {
+                IPAddress address = new IPAddress(encodedIPAddress);
+                return address.ToString();
+            }
+            catch
+            {
+                throw new CryptographicException("Certificate contains invalid IP address.");
+            }
+        }
+
+#if NETSTANDARD2_1
+        /// <summary>
+        /// Encode the Subject Alternative name extension.
+        /// </summary>
+        private byte[] Encode()
+        {
+            var sanBuilder = new SubjectAlternativeNameBuilder();
+            foreach (var uri in m_uris)
+            {
+                sanBuilder.AddUri(new Uri(uri));
+            }
+            EncodeGeneralNames(sanBuilder, m_domainNames);
+            EncodeGeneralNames(sanBuilder, m_ipAddresses);
+            var extension = sanBuilder.Build();
+            return extension.RawData;
+        }
+
+        /// <summary>
+        /// Encode a list of general Names in a SAN builder.
+        /// </summary>
+        /// <param name="sanBuilder">The subject alternative name builder</param>
+        /// <param name="generalNames">The general Names to add</param>
+        private void EncodeGeneralNames(SubjectAlternativeNameBuilder sanBuilder, IList<string> generalNames)
+        {
+            foreach (string generalName in generalNames)
+            {
+                IPAddress ipAddr;
+                if (String.IsNullOrWhiteSpace(generalName))
+                {
+                    continue;
+                }
+                if (IPAddress.TryParse(generalName, out ipAddr))
+                {
+                    sanBuilder.AddIpAddress(ipAddr);
+                }
+                else
+                {
+                    sanBuilder.AddDnsName(generalName);
+                }
+            }
+        }
+#else  
+        /// <summary>
+        /// Encode the Subject Alternative name extension.
+        /// </summary>
+        private byte[] Encode()
+        {
+            return BouncyCastle.X509Extensions.BuildSubjectAltNameExtension(m_uris, m_domainNames, m_ipAddresses).RawData;
```

Comment:
This isn't too hard with AsnWriter.

Reply:
Yes, I have sanbuilder on .NET Core 3 and netcore 2/net462 may be deprecated soon...not worth the extra effort.